Overhaul "make doc"

fingolfin opened the pull request:
* unified the `makedocrel.g` and `makedocreldata.g` for all help books (and added them in the case of the developer book)
Changes affecting `make doc`:
* reduces number of GAPDoc invocations for each book from 4 to 3 (the black&white manual is only built once, in the second run)
* now also builds the dev manual
* does not use the makedocrel.g files anymore
* does not require IO packages anymore

alex-konovalov commented on doc/dev/main.xml:
```diff
@@ -9,7 +9,7 @@
 >
 
-<Book Name="Development">
+<Book Name="dev">
```
Makes sense - e.g. we use `<Book Name="tut">` in Tutorial etc. Good when book name includes in cross-references.

alex-konovalov commented on doc/changes/makedocreldata.g:
```diff
@@ -9,7 +9,8 @@ GAPInfo.ManualDataChanges:= rec(
   pathtoroot:= "../..",
 
-  files:= [ ],
+  files:= [
```
I guess this change is here to use the same style as in other `doc/*/makedocreldata.g` files, so that in the future any files containing documentation sections will be added between these two lines.

fingolfin replied:
Actually, I only did it to reduce the diffs to the other `makedocreldata.g` files. Other than that, i think we can and should mostly get rid of these files...

alex-konovalov commented on doc/make_doc.in:
```diff
@@ -3,32 +3,57 @@
-base:="$abs_top_srcdir";;
-books:=["ref", "tut", "changes", "hpc"];;
+base:="@abs_top_srcdir@";;
+books:=["ref", "tut", "changes", "hpc", "dev"];;
```
I don't think we are going to include doc/dev manual in GAP 4.9, so I suggest that one should not attempt to build it if the directory doc/dev is not there. This will ensure that we are able to build manuals during release wrapping.

fingolfin replied:
I updated this PR to skip over any missing directories.

But I am not sure why we wouldn't include `doc/dev` with GAP 4.9? Perhaps because it is incomplete and out of date? That would make some sense, I guess; but I hope future versions will be shipped.

alex-konovalov replied:
Exactly because of it being incomplete and out of date. I've started a revision attempt in PR #506 but it was not revisited for a while...

fingolfin commented:
Update this PR, and added two new commits which slightly tweak the `dev` manual. I can also split those off into a separate PR if desired, but it didn't seem worth it.

alex-konovalov commented:
Should be fine to merge this. I can rebase #506 after that (and probably we have to merge #506 then, and make incremental changes after that...)

fingolfin commented:
@alex-konovalov well, then feel free to merge this, and rebase #506 :-)
